mi/dataset/driver/zplsc_c/zplsc_c_echogram.py

```diff
@@ -0,0 +1,300 @@
+figure instead of at the bottom of it.
+"""
+
+from datetime import datetime
+import numpy as np
+import matplotlib
+from matplotlib.dates import date2num, num2date
+from modest_image import imshow
+import mi.dataset.driver.zplsc_c.zplsc_functions as zf
+
+matplotlib.use("Agg")
+import matplotlib.pyplot as plt
+
+__author__ = 'Craig Risien/Rene Gelinas'
+__license__ = 'Apache 2.0'
+
+REF_TIME = date2num(datetime(1900, 1, 1, 0, 0, 0))
+
+
+class ZplscCParameters(object):
+    # TODO: This class should be replaced by methods to get the CCs from the system.
+    # Configuration Parameters
+    Salinity = 32   # Salinity in psu
+    Pressure = 150  # in dbars (~ depth of instrument in meters).
+    Bins2Avg = 1    # number of range bins to average - 1 is no averaging
+
+
+class ZplscCCalibrationCoefficients(object):
+    # TODO: This class should be replaced by methods to get the CCs from the system.
+    ka = 464.3636
+    kb = 3000.0
+    kc = 1.893
+    A = 0.001466
+    B = 0.0002388
+    C = 0.000000100335
+
+    TVR = []
+    VTX = []
+    BP = []
+    EL = []
+    DS = []
+
+    # Freq 38kHz
+    TVR.append(1.691999969482e2)
+    VTX.append(1.533999938965e2)
+    BP.append(8.609999902546e-3)
+    EL.append(1.623000030518e2)
+    DS.append(2.280000038445e-2)
+
+    # Freq 125kHz
+    TVR.append(1.668999938965e2)
+    VTX.append(5.8e+01)
+    BP.append(1.530999969691e-2)
+    EL.append(1.376999969482e2)
+    DS.append(2.280000038445e-2)
+
+    # Freq 200kHz
+    TVR.append(1.688999938965e2)
+    VTX.append(9.619999694824e1)
+    BP.append(1.530999969691e-2)
+    EL.append(1.456000061035e2)
+    DS.append(2.250000089407e-2)
+
+    # Freq 455kHz
+    TVR.append(1.696000061035e2)
+    VTX.append(1.301000061035e2)
+    BP.append(8.609999902546e-3)
+    EL.append(1.491999969482e2)
+    DS.append(2.300000004470e-2)
+
+
+class ZPLSCCPlot(object):
+    font_size_small = 14
+    font_size_large = 18
+    num_xticks = 25
+    num_yticks = 7
+    interplot_spacing = 0.1
+    lower_percentile = 5
+    upper_percentile = 95
+
+    def __init__(self, data_times, channel_data_dict, frequency_dict, depth_range, _min_db=None, _max_db=None):
+        self.fig = None
+        self.power_data_dict = self._transpose_and_flip(channel_data_dict)
+
+        if (_min_db is None) or (_max_db is None):
+            self.min_db, self.max_db = self._get_power_range(channel_data_dict)
+        else:
+            self.min_db = _min_db
+            self.max_db = _max_db
+
+        self.frequency_dict = frequency_dict
+
+        # convert ntp time, i.e. seconds since 1900-01-01 00:00:00 to matplotlib time
+        self.data_times = (data_times / (60 * 60 * 24)) + REF_TIME
+        max_depth, _ = self.power_data_dict[1].shape
+        self._setup_plot(depth_range, max_depth)
+
+    def generate_plots(self):
+        """
+        Generate plots for all channels in data set
+        """
```

Comment:
This is out of the scope of this ticket.  Ticket #12866 was opened to track this issue.  It will be done when the other B-Series Echogram tickets are worked.